packages/web-components/fast-element/docs/api-report.md

```diff
@@ -171,7 +171,8 @@ export class Controller extends PropertyChangeNotifier {
     readonly element: HTMLElement;
     emit(type: string, detail?: any, options?: Omit<CustomEventInit, "detail">): void | boolean;
     static forCustomElement(element: HTMLElement): Controller;
-    readonly isConnected: boolean;
```

Comment:
We don't want `isConnected` publicly assignable do we? Should we shadow this so there is only a public getter?

Reply:
What do you think is the best way to accomplish that from a TS perspective?

Reply:
It would involve a third property, but I think you could use an IIFE to hide it from the private API as well. I think this would work.

```ts
private _isConnected: boolean;
public get isConnected(): boolean {
    return this._isConnected;
}
constructor() {
    // other stuff
        (() => {
            let connected = false;
            Object.defineProperty(this, "_isConnected", {
                get() {
                    Observable.track(this, "isConnected");
                    return connected
                },
                set(value: boolean) {
                    Observable.notify(this, "isConnected");
                    connected = value;
                },
            })
        })()
}
```

Reply:
That makes me feel a bit uncomfortable. I'm wondering if I can use `@internal` on a setter only? TS wouldn't let me make the property setter private...but I think they enabled that in a later version of TS. Let me play around with it a bit.

Reply:
Nope...doesn't look like the `@internal` trick will work.

Reply:
I think we are trying too hard 🤣  How about this?

```ts
public get isConnected(): boolean {
  Observable.track(this, "isConnected");
  return this._isConnected;
}

private setIsConnected(value: boolean) {
  this._isConnected = value;
  Observable.notify(this, "isConnected");
 }
```

Reply:
I'm good with whatever so long as we're not saying `isConnected` is publicly mutable :)